python/utils/__init__.py

```diff
@@ -1,5 +1,9 @@
 from utils import colorText, str2bool, find_moose_executable, runExe, check_configuration
-from MooseYaml import MooseYaml
+
+try:
+    from MooseYaml import MooseYaml
+except:
+    pass
```

Comment:
Right now the test harness is using the "utils" module (which we should
rename to "mooseutils"). That is, there is an "import utils" statement.

This PR adds a new utility that requires pyyaml (MooseYaml has "import
yaml").

Thus, the TestHarness now requires yaml, which on falcon isn't all that
easy. This statement will allow the test harness run without pyyaml, but if
pyyaml is available (i.e., with our package) then you get this extra
capability. You shouldn't be running peacock or moosedocs scripts from
falcon, so why require a python module that you shouldn't/can't use.

This PR also adds a utility to check your configuration, which I would like
to start moving towards. This will allow scripts to check that the python
environment is correct while keeping the dependencies for things at a
minimum.

On Tue, Aug 2, 2016 at 8:52 AM, Cody Permann <notifications@github.com>
wrote:

> In python/utils/__init__.py
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_7455-23discussion-5Fr73170197&d=CwMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=S7dclt_6ZnF9B1Q6JbGnj5gaDLzgIx2IpWStCVvZYjA&s=grt_u8NDLVWEyCoUrlkYl0_eN0qlljGK_bOxwHxfqN8&e=>
> :
>
> > @@ -1,5 +1,9 @@
> >  from utils import colorText, str2bool, find_moose_executable, runExe, check_configuration
> > -from MooseYaml import MooseYaml
> > +
> > +try:
> > +    from MooseYaml import MooseYaml
> > +except:
> > +    pass
>
> I guess I don't understand how this is optional. Maybe if it's not there
> we set a Boolean and report a better error if we attempt to use the library?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_7455_files_567dab45c376444a7452baed40178086af4b9ab6-23r73170197&d=CwMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=S7dclt_6ZnF9B1Q6JbGnj5gaDLzgIx2IpWStCVvZYjA&s=xL06_NByOyQNaGDxyfEQHG_l9Clagso6RQcl7eu7Er0&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAU1Vbyde1K-2D-5FU3iY8lGvCCR60INhzKCks5qb1mugaJpZM4JZw2w&d=CwMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=S7dclt_6ZnF9B1Q6JbGnj5gaDLzgIx2IpWStCVvZYjA&s=WLTDmsB4zS9tU2fgiBJD3WCMMeMp4kkUh-pjpfitDQA&e=>
> .
>